src/runtime-rs/crates/runtimes/common/src/container_manager.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::types::{
+    ContainerConfig, ExecProcessRequest, KillRequest, ProcessExitStatus, ProcessID,
+    ProcessStateInfo, ResizePtyRequest, ShimPid, ShutdownRequest, StatsInfo, UpdateRequest,
+};
+
+#[async_trait]
+pub trait ContainerManager: Send + Sync {
+    // container lifecycle
+    async fn create_container(&self, config: ContainerConfig) -> Result<u32>;
+    async fn pause_container(&self, cid: &str) -> Result<()>;
+    async fn resume_container(&self, cid: &str) -> Result<()>;
+    async fn stats_container(&self, cid: &str) -> Result<StatsInfo>;
+    async fn update_container(&self, req: UpdateRequest) -> Result<()>;
+
+    // process lifecycle
+    async fn close_io_process(&self, process_id: &ProcessID) -> Result<()>;
+    async fn delete_process(&self, process_id: &ProcessID) -> Result<ProcessStateInfo>;
+    async fn exec_process(&self, req: ExecProcessRequest) -> Result<()>;
+    async fn kill_process(&self, req: &KillRequest) -> Result<()>;
+    async fn resize_pty_process(&self, req: &ResizePtyRequest) -> Result<()>;
+    async fn start_process(&self, process_id: &ProcessID) -> Result<u32>;
```

Comment:
nit: choose `cid` and `pid` or `container_id` and `process_id`

Reply:
Thanks for your review, rename `u32` to  `ShimPid`